src/AutoRest.CSharp/Mgmt/Decorator/ResourceIdentifierChooser.cs

```diff
@@ -16,11 +17,17 @@ namespace AutoRest.CSharp.Mgmt.Decorator
 {
     internal static class ResourceIdentifierChooser
     {
-        internal static Type GetResourceIdentifierType(this OperationGroup operation, MgmtObjectType mgmtObjectType, MgmtConfiguration config)
+        internal static Type GetResourceIdentifierType(this OperationGroup operationGroup, BuildContext context)
         {
-            if (operation.ParentResourceType(config) == ResourceTypeBuilder.Subscriptions)
+            var config = context.Configuration.MgmtConfiguration;
+            //Go up until we get to the operation group level that is directly under a resource group, subscription or tenant.
+            while (operationGroup.ParentOperationGroup(context) != null)
+            {
+                operationGroup = operationGroup.ParentOperationGroup(context)!;
+            }
```

Comment:
This is different from getting parents of an ID, in which the ultimate parent is always a tenant. Here the ultimate `parentOperationGroup` is `null` in the case when `operationGroup` is a child of either a resource group, subscription or tenant as we look up for the `parentOperationGroup` in the codemodel of an RP which does not contain the model of the above three.